pandas/core/generic.py

```diff
@@ -7848,13 +7848,8 @@ def interpolate(
             if np.any(obj.dtypes == object):
                 # GH#53631
                 if not (obj.ndim == 2 and np.all(obj.dtypes == object)):
```

Comment:
Could you combine these conditionals? `np.any` and `np.all` overlap

Reply:
thank you for the comment. I combined two `if` statement into one.